src/Sylius/Bundle/ShopBundle/EventListener/UserRegistrationListener.php

```diff
@@ -82,4 +117,18 @@ private function handleUserVerificationToken(ShopUserInterface $user)
 
         $this->eventDispatcher->dispatch(UserEvents::REQUEST_VERIFICATION_TOKEN, new GenericEvent($user));
     }
+
+    /**
+     * @param ShopUserInterface $user
+     */
+    private function verifyAndLogin(ShopUserInterface $user)
+    {
+        $user->setVerifiedAt(new \DateTime());
```

Comment:
IMO this feature should only enable the user account and allow for a login, but not automatically verify the email address. The verification email should be sent anyway, so the user can confirm that the email address belongs to him.